src/core/lv_obj.h

```diff
@@ -138,6 +142,48 @@ typedef _lv_obj_flag_t lv_obj_flag_t;
 typedef uint32_t lv_obj_flag_t;
 #endif /*DOXYGEN*/
 
+#if LV_USE_OBJ_PROPERTY
+enum {
+    /* obj property of flags */
+    LV_PROPERTY_ID(OBJ, FLAGS_START,            LV_PROPERTY_TYPE_INT,       1),
```

Comment:
I wonder if we can make it easier to maintain and automatic to follow the flag changes. 

Can we say that flag start is `0x8000 0000` (1 << 31) and the values are e.g. `0x80000000 | LV_OBJ_FLAG_HIDDEN`?

An other option is to add a unit test for each flag so that we can notice if there are any inconsistencies. Adding unit test seems more elegant.

What do you think?

Reply:
I think I have overlooked it. We can simply regard `flags` as one property.
Set/get will modify/read obj->flags directly with property->value. User should read/modify/write the flags.

I'll update the commit to see if that works more clearly.

Reply:
I have added it in https://github.com/lvgl/lvgl/pull/4655
The property `FLAGS` can be used to remove or add flags based on bit31 in value, using the way you suggested.